its/it/runner.ts

```diff
@@ -16,5 +31,23 @@ export async function main() {
     if (!lastAnalysisDate || lastAnalysisDate === previousLastAnalysisDate) {
       throw new Error("Analysis date did not change");
     }
-  }
+}
+
+export async function main() {
+  const azdoApi = getAzdoApi();
+
+  const executionPlan = getExecutionPlan(testCases);
+  console.log(`Execution plan: \n${executionPlan.map(testCases => testCases.map(testCase => testCase.pipelineName)).join("\n")}`);
+
+  console.log("Running all test cases");
+  await Promise.all(
+    executionPlan
+      .map(testCases => {
+        return (async () => {
+          for (const testCase of testCases) {
+            await run(azdoApi, testCase);
```

Comment:
👏🏻

Reply:
One small thing, the azdo logs are mixed
https://dev.azure.com/sonarsource/SonarScannerAzdo/_build/results?buildId=103733&view=logs&j=f894ae3e-d562-54f6-0b47-95efd0b9d380&t=72cc528c-816f-5251-a869-6fa6df3d61ce

Though at this point we can probably ignore them. But debugging a failed pipeline will be harder. 

We could cleanup the logs, OR prefix them with the buildId?

```
Getting last analysis date for sonarsource-sonar-scanner-azdo-it_dummy-project-cli at https://sonarcloud.io/api/components/show?component=sonarsource-sonar-scanner-azdo-it_dummy-project-cli&branch=task/br/sonarazdo-423-parallelize-it-pipelines-execution&ps=1...
Running pipeline "pipeline-sonarcloud-v3-dotnet-8.0.3.99785-windows" on branch "task/br/sonarazdo-423-parallelize-it-pipelines-execution"
Running pipeline "pipeline-sonarcloud-v3-cli-embedded-unix" on branch "task/br/sonarazdo-423-parallelize-it-pipelines-execution"
Running pipeline "pipeline-sonarcloud-v3-dotnet-8.0.3.99785-unix" on branch "task/br/sonarazdo-423-parallelize-it-pipelines-execution"
Running pipeline "pipeline-sonarcloud-v3-other-gradle-unix" on branch "task/br/sonarazdo-423-parallelize-it-pipelines-execution"
Running pipeline "pipeline-sonarcloud-v3-other-maven-unix" on branch "task/br/sonarazdo-423-parallelize-it-pipelines-execution"
Build 103734 queued
Build 103735 queued
Build 103736 queued
Build 103737 queued
##[debug]Agent environment resources - Disk: / Available 20102.45 MB out of 74244.74 MB, Memory: Used 1353.00 MB out of 6921.00 MB, CPU: Usage 80.46%
Build 103738 queued
##[debug]Agent environment resources - Disk: / Available 20102.45 MB out of 74244.74 MB, Memory: Used 1356.00 MB out of 6921.00 MB, CPU: Usage 6.74%
##[debug]Agent environment resources - Disk: / Avai
```